libgadget/blackhole.c

```diff
@@ -705,20 +705,23 @@ void blackhole_make_one(int index) {
     if(P[index].Type != 0)
         endrun(7772, "Only Gas turns into blackholes, what's wrong?");
 
-    int child = slots_fork(index, 5);
+    int child;
 
-    BHP(child).FormationTime = All.Time;
-    /*Ensure that mass is conserved*/
-    double BHmass = All.SeedBlackHoleMass;
+    /*If the particle mass is less than that needed for a black hole, convert.*/
     if(P[index].Mass <= All.SeedBlackHoleMass) {
-        slots_mark_garbage(index);
-        BHmass = P[index].Mass;
+        child = slots_convert(index, 5);
+    }
+    /*Otherwise create a new particle*/
+    else {
+        child = slots_fork(index, 5);
+        /*Ensure that mass is conserved*/
+        P[child].Mass = All.SeedBlackHoleMass;
+        P[index].Mass -= All.SeedBlackHoleMass;
     }
-    P[child].Mass = BHmass;
-    P[index].Mass -= BHmass;
     BHP(child).base.ID = P[child].ID;
-    BHP(child).Mass = BHmass;
+    BHP(child).Mass = P[child].Mass;
```

Comment:
It is important to set the Mass to SeedBlackHoleMass.

BHP().Mass does not agree with the gravity mass in P.

Reply:
line 712 would have set it to P[].Mass

Reply:
Added comment

Reply:
Are you sure? That's not what it does in the master branch. In both branches the logic should be the same: It is min(P[parent].Mass, SeedBlackHoleMass), which happens to also be the same as P[child].Mass.

Reply:
I am sure. The master has a bug then. The older version was correct.

https://github.com/MP-Gadget/MP-Gadget/blame/f5a6feaade9a53269ce073aaaefee655993c0f51/blackhole.c

SeedBlackholeMass is the accretion model which has almost nothing to do with the gravity mass of the object. lower seed mass produces later Eddington-limited accretion / bright quasar phases. If we give the blackholes different seed masses it would mean we are by chance using different accretion models!

Reply:
Ah, indeed. This was introduced by  b5d1d95f775b7e7d9be248328d0da565f94d5393
which limited the BlackHole to only being as massive as the seed particle, but it should apply only to the gravitational mass, not the accretion mass.